File: server/users-microservice/src/utils/TokenController.py
from datetime import datetime

import uuid
import jwt

from config.config import jwtSecret

class TokenController:
    
    def generateToken(name, email):
        token_identifier = str(uuid.uuid4()) + str(uuid.uuid4());

        payload = {
            'iss': 'users_microservice',
            'sub': 'user_token',
            'csrf': token_identifier,
            'iat': datetime.now(),
            'userName': name,
            'userEmail': email,
        }

        token = jwt.encode(payload, jwtSecret, 'HS256')

        csrf =  {
            'csrf': token_identifier
        }

        return {
            'token': token,
            'csrf': csrf
        }

    def decodeToken(token):
        options = {
            'verify_exp': False,
            'verify_aud': False,
            'verify_iat': False
        }

        return jwt.decode(token, key=jwtSecret, options=options)

File: server/users-microservice/src/utils/tokenController.js
'use strict';

import jwt from 'jsonwebtoken';
import uuid from 'uuid';

import { jwtSecret } from '../config/config';

export const generateToken = (name, email, id) => {
  let tokenIdentifier = uuid.v4() + uuid.v4();

  let payload = {
    iss: 'users_microservice',
    sub: 'user_token',
    csrf: tokenIdentifier,
    iat: Math.floor(Date.now() / 1000) - 30 ,
    userName: name,
    userEmail: email,
    userId: id
  };

  let token = jwt.sign(payload, jwtSecret);

  let csrf =  {
    'csrf': tokenIdentifier
  };

  return {
    token,
    csrf
  };
};

export const decodeToken = token => {
  try {
    return jwt.verify(token, jwtSecret);
  } catch(err) {
    
  }
};